classes/class.ilMumieTaskGradeListGUI.php

```diff
@@ -0,0 +1,207 @@
+
+        $this->setFormName('participants');
+
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_submission_date'), 'date');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_use_grade'), 'useGrade');
+        $this->addColumn($lng->txt('rep_robj_xmum_frm_list_grade'), 'grade');
+        
+        $this->tpl->addBlockFile(
+            "TBL_CONTENT",
+            "tbl_content",
+            "tpl.mumie_grade_list.html",
+            "Customizing/global/plugins/Services/Repository/RepositoryObject/MumieTask"
+        );
+        if ($parentObj->object->getPrivateGradepool() != -1) {
+            $gradesync  = new  ilMumieTaskGradeSync($parentObj->object, false);
+            $xGrades = $gradesync->getAllXapiGradesByUser();
+            $syncId = $gradesync->getSyncIds(array($user_id))[0];
+            if (!empty($xGrades)) {
+                foreach ($xGrades as $xGrade) {
+                    if ($xGrade->actor->account->name == $syncId) {
+                        $this->tpl->setCurrentBlock("tbl_content");
+                        $this->css_row = ($this->css_row != "tblrow1")
+                            ? "tblrow1"
+                            : "tblrow2";
+
+                        $this->tpl->setVariable("CSS_ROW", $this->css_row);
+                        $this->tpl->setVariable("VAL_GRADE", round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'user_id', $user_id);
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'newGrade', round($xGrade->result->score->raw * 100));
+                        $this->ctrl->setParameterByClass('ilObjMumieTaskGUI', 'timestamp', strtotime($xGrade->timestamp));
+
+                        $this->tpl->setVariable("LINK_NAME", $this->ctrl->getLinkTarget($parentObj, 'displayGradeList'));
+
+                        $this->tpl->setVariable("LINK_TXT", $lng->txt('rep_robj_xmum_frm_list_use_grade'));
+                        $this->tpl->setVariable("VAL_DATE", substr($xGrade->timestamp, 0, 10) . " - " . substr($xGrade->timestamp, 11, 5));
+                        $this->tpl->setCurrentBlock("tbl_content");
+                        $this->tpl->parseCurrentBlock();
+                    }
+                }
+            }
+        }
+
+        $this->enable('header');
+        $this->enable('sort');
+        $this->setEnableHeader(true);
+    }
+
+    public static function overrideGrade($parentObj)
+    {
+        global $ilDB, $DIC, $lng;
+        $percentage = $_GET['newGrade'];
+        $DIC->database()->update(
+            'ut_lp_marks',
+            array(
+                "status_changed" => array('text', date("Y-m-d H:i:s", $_GET['timestamp'])),
+                "mark" => array('int', $percentage),
+            ),
+            array(
+                'obj_id' => array('int', $parentObj->object->getId()),
+                'usr_id' => array('int', $_GET['user_id']),
+            )
+        );
```

Comment:
We should always reuse code if possible. This looks rather similar to `ilMumieTaskLPStatus::upsertMarks`

Reply:
done